js/ui/components/clubhouse.js

```diff
@@ -291,7 +291,15 @@ var ClubhouseComponent = new Lang.Class({
         this._clubhouseButtonManager = new ClubhouseButtonManager();
         this._clubhouseButtonManager.connect('open-clubhouse', () => {
             let time = global.get_current_time();
-            this.show(time);
+
+            // When the Shell is restarted, we cannot really get the Clubhouse window actor
+            // if the Clubhouse was running, which means the close button won't be shown;
+            // thus, the 'toggle' action below allows users to at least hide the Clubhouse
+            // window and reopen it, which would fix the issue
```

Comment:
It feels like we should spend some time trying to understand why there are no window actors yet by the time this component is initialized, and how to defer initialization until the windows are known to the compositor, instead of putting this hack here.

Reply:
Okay. I will try the solution I have in mind. Even if I fix the issue, this logic is harmless so we can consider keeping it to prevent eventual problems of having the Clubhouse button shown when the Clubhouse is also being shown (as if that happens there's no way to close the Clubhouse, making it a bigger issue than having the wrong button being shown).